CONTRIBUTING.md

```diff
@@ -168,12 +168,58 @@ These are garage-specific rules which are not part of the aforementioned style g
 
-Additional standards:
+### Docstrings
 * Docstrings for `__init__` should be included in the class docstring as suggested in the [Google example](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
 * Docstrings should provide full type information for all arguments, return values, exceptions, etc. according to the Google format
+* When documenting fields which are numpy arrays or other tensor types (and collections thereof), please carefully document the expected input shape of the field. See below for shape conventions.
+* For shapes and equations, use the Sphinx `:math:` directive to render them properly with mathematical symbols. 
+
+#### Conventions for documenting tensor shapes
+Data which include a meaningful time-series dimension (e.g. trajectories) should always document that dimension explicitly, even if that dimension has been flattened out. Data containing only non time-series samples should omit the time dimension.
+
+Always use the Sphinx `:math:` directive to render your shapes properly.
+
+| Symbol    | Description                                                                                             |
+|-----------|---------------------------------------------------------------------------------------------------------|
+| `(...)`   | Tensor shapes are enclosed in parentheses, e.g a batch of `(N, S^*)` samples                            |
+| `N`       | Batch dimension (e.g. trajectories or samples)                                                          |
+| `T`       | Time dimension                                                                                          |
+| `S^*`     | Variadic parts of a tensor shape, which will be broadcast or ignored                                    |
+| `[.]`     | Variable-length dimensions are enclosed in square brackets, e.g. `[K]` if `K` is the dimension variable |
+| `\bullet` | Flattening operator, e.g. `(N \bullet T, K)` has a flattened first dimension of length `N * T`          |
+
+**Example**
+```python
+def concatenate_time(paths):
+    """Concatenate a list of variable-length tensors along the time dimemsion.
+    
+    Concatenates a list `paths` of `N` variable-length time-series tensors
+    along their time dimension, producing a single time-series tensor with the
+    component tensors arranged along a single batch dimension.
+    
+    Args:
+        paths (list[numpy.ndarray]): A list of :math:`N` tensors to combine
+            into a single batch of tensors, with elements of shape
+            :math:`([T], S^*)`
+            
+    Returns:
+        numpy.ndarray: Time-flattened version of `paths`, with shape
+            :math:`(N \bullet [T], S^*)`
+    """
+```
```

Comment:
Should we also list out the keys and values / only the keys if paths is a dictionary? I think that's usually the case in garage. If so, I believe it would be helpful to illustrate how to render that nicely, e.g.

- Keys and values:
```
...
    Args:
        paths (dict[list]): A dictionary with values of shape :math:`(N \bullet [T], S^*)`.

    Note:
        The returned paths contain the following keys and values are of shape:
        * `observation`: :math:`(N \bullet [T], observation_dim^*)`
        * `action`: :math:`(N \bullet [T], action_dim^*)`
        * `reward`: :math: `(N \bullet [T])`
...
```

Only keys:
```
...
    Args:
        paths (dict[list]): A dictionary with values of shape :math:`(N \bullet [T], S^*)`.

    Note:
        The returned paths contain the keys: `observation`, `action`, `reward`...
...
```

Reply:
Yes. Feel free to amend this branch :)

Reply:
Which one do you think it's better? (I think it's always better to be more detail, but sometimes it might just be an overkill)

Reply:
the first one, assuming it renders nicely in Sphinx

you should just use
```python
        * `observation`: :math:`(N \bullet [T], S^*)`
        * `action`: :math:`(N \bullet [T], S^*)`
```

the words won't render nicely in Sphinx, plus it's obvious from the context what `S^*` should be.

Reply:
I will let you send the PR to update this @ahtsan